azure-pipelines-e2e-tests-template.yml

```diff
@@ -107,45 +119,47 @@ stages:
+            unzip Microsoft.Spark.Worker-${{ parameters.backwardCompatibleRelease }}.zip -d $([System.IO.Directory]::GetParent($env:BACKWARD_COMPATIBLE_DOTNET_WORKER_DIR).FullName)
+
+            if ($env:AGENT_OS -eq 'Linux') {
+              chmod +x "${env:BACKWARD_COMPATIBLE_DOTNET_WORKER_DIR}${env:PATH_SEPARATOR}Microsoft.Spark.Worker"
+            }
+          condition: eq( ${{ test.disableCompatTests }}, false)
+          displayName: 'Setup Backward Compatible Microsoft Spark Worker ${{ parameters.backwardCompatibleRelease }}'
+          workingDirectory: $(Build.BinariesDirectory)
+
+        - task: DotNetCoreCLI@2
+          displayName: 'E2E Backward Compatibility Tests'
+          inputs:
+            command: test
+            projects: '**/Microsoft.Spark*.E2ETest/*.csproj'
+            arguments: '--configuration $(buildConfiguration) ${{ option.backwardCompatibleTestOptions }}'
+            workingDirectory: $(Build.SourcesDirectory)$(PATH_SEPARATOR)dotnet-spark
+          env:
+            HADOOP_HOME: $(Build.BinariesDirectory)$(PATH_SEPARATOR)hadoop
+            SPARK_HOME: $(Build.BinariesDirectory)$(PATH_SEPARATOR)spark-${{ test.version }}-bin-hadoop2.7
+            DOTNET_WORKER_DIR: $(BACKWARD_COMPATIBLE_DOTNET_WORKER_DIR)
 
         - checkout: forwardCompatibleRelease
           path: s$(PATH_SEPARATOR)dotnet-spark-${{ parameters.forwardCompatibleRelease }}
 
-        # - task: Maven@3
-        #   displayName: 'Maven build src for forward compatible release v${{ parameters.forwardCompatibleRelease }}'
-        #   inputs:
-        #     mavenPomFile: $(Build.SourcesDirectory)$(PATH_SEPARATOR)dotnet-spark-${{ parameters.forwardCompatibleRelease }}$(PATH_SEPARATOR)src$(PATH_SEPARATOR)scala$(PATH_SEPARATOR)pom.xml
-
-        # - task: DotNetCoreCLI@2
-        #   displayName: 'E2E Forward Compatibility Tests'
-        #   inputs:
-        #     command: test
-        #     projects: '**/Microsoft.Spark*.E2ETest/*.csproj'
-        #     arguments: '--configuration $(buildConfiguration) ${{ option.forwardCompatibleTestOptions }}'
-        #     workingDirectory: $(Build.SourcesDirectory)$(PATH_SEPARATOR)dotnet-spark-${{ parameters.forwardCompatibleRelease }}
-        #   env:
-        #     HADOOP_HOME: $(Build.BinariesDirectory)$(PATH_SEPARATOR)hadoop
-        #     SPARK_HOME: $(Build.BinariesDirectory)$(PATH_SEPARATOR)spark-${{ test.version }}-bin-hadoop2.7
-        #     DOTNET_WORKER_DIR: $(CURRENT_DOTNET_WORKER_DIR)
\ No newline at end of file
+        - task: Maven@3
+          displayName: 'Maven build src for forward compatible release v${{ parameters.forwardCompatibleRelease }}'
+          condition: eq( ${{ test.disableCompatTests }}, false)
+          inputs:
+            mavenPomFile: $(Build.SourcesDirectory)$(PATH_SEPARATOR)dotnet-spark-${{ parameters.forwardCompatibleRelease }}$(PATH_SEPARATOR)src$(PATH_SEPARATOR)scala$(PATH_SEPARATOR)pom.xml
+
+        - task: DotNetCoreCLI@2
```

Comment:
Should this task even be attempted if test.disableCompatTests is true ?  Also do we need separate flags for both backward/forward compat tests ?

Reply:
Added separate flags.